app/javascript/vue/shared/datatable/renderers/favorite.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <div v-if="hasFavorite">
+    <button @click="updateFavorite(!favorite)"
+            class="flex justify-center items-center w-full h-9 bg-transparent border-none cursor-pointer">
+      <div v-if="favorite" class="sn-icon sn-icon-star-filled text-sn-alert-brittlebush"></div>
+      <div v-else @mouseover="hovered = true"
+                  @mouseleave="hovered = false">
+        <div v-if="hovered" class="text-sn-grey-500 sn-icon sn-icon-star-filled"></div>
```

Comment:
this can probably be done using a tailwind class, should be something like hover:sn-icon-star-filled